iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/api/bicep/TypeReferenceSuffix.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.api.bicep;
+
+/**
+ * Interface for type suffix that build into {@link TypeExpressionAble}.
+ */
+public interface TypeReferenceSuffix {
+
+  TypeExpressionAble setType(TypeExpressionAble type);
```

Comment:
Do we want to make this method available in the public API?
Maybe it's not better, but I may have seen sometimes that we keep such a method in the implementation only and cast the interface to the implementation in the `TreeFactory` to access this method.
Otherwise, as we discussed, there is still the solution of having a builder pattern, but it adds some overheads.

Reply:
I guess having an implicit expected method on implementations and then casting is definitely not something I want to do, the OOP experts would kill me for it

I guess suffixes are kind of builders already? They just aren't named builders explicitly (personally prefer it this way), but they are not part of the AST (since they don't inherit `ArmTree`). Maybe it's just not in the right package then? But unsure where I should put them 🤔